tests/spec_decode/e2e/conftest.py

```diff
@@ -203,16 +203,26 @@ def maybe_assert_ngram_worker(llm):
 
 
 def get_output_from_llm_generator(
-        llm_generator, prompts,
-        sampling_params) -> Tuple[List[str], List[List[int]]]:
+    llm_generator,
+    prompts,
+    sampling_params,
+    ensure_gpu_advance_used: bool = False,
+) -> Tuple[List[str], List[List[int]]]:
     tokens: List[str] = []
     token_ids: List[List[int]] = []
     for llm in llm_generator():
         maybe_assert_ngram_worker(llm)
 
         outputs = llm.generate(prompts, sampling_params, use_tqdm=True)
+
         token_ids = [output.outputs[0].token_ids for output in outputs]
         tokens = [output.outputs[0].text for output in outputs]
+
+        if ensure_gpu_advance_used:
+            num_gpu_runs = (llm.llm_engine.model_executor.driver_worker.
+                            proposer_worker.model_runner._num_gpu_runs)
+            assert num_gpu_runs > 0
```

Comment:
It's generally not a good idea to add a field only for testing. Is there any tricky part to leverage mock and injection? I could help with a unit test if needed.

Reply:
Yeah would be good if you could add mock/injection, I'm not sure how to do it. The reason I used _tests/spec_decode/e2e/test_multistep_correctness.py::test_spec_decode_e2e_greedy_correctness_tiny_model_bs1_ is because it verifies full e2e correctness.

Reply:
Let me work on one for you soon.